cluster-autoscaler/cloudprovider/azure/azure_config.go

```diff
@@ -387,6 +405,17 @@ func (cfg *Config) getAzureClientConfig(authorizer autorest.Authorizer, env *azu
 		}
 	}
 
+	// A proxy service is required to access resources for the managed system pool within automatic clusters.
```

Comment:
micro nit: let's move this up one block, before we check for the ExtendedLocation config, so that all of our "override from default azClientConfig property settings"  code is organized together

Reply:
Updated!